test/save_assemblies_from_fastas_integration_test.py

```diff
@@ -576,4 +584,171 @@ def test_fail_mismatched_inputs(config, impl, context, scratch):
+    ]
+
+    # copy 2 assembly files into the data dir
+    for file_name in file_names:
+        shutil.copy(data / file_name, tmp_dir)
+
+    params = {
+        'workspace_id': config['ws_id'],
+        'inputs': [
+            {'file': tmp_dir / 'GCA_002506415.1_ASM250641v1_genomic.fna.gz', 'assembly_name': 'GCA_002506415.1_ASM250641v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000007065.1_ASM706v1_genomic.fna.gz', 'assembly_name': 'GCF_000007065.1_ASM706v1_genomic.fna.gz'}
+        ]
+    }
+
+    var_name = "MAX_THREADS"
+    config['file_config']["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "10e"
+    impl = AssemblyUtil(config['file_config'])
+    with raises(Exception) as got:
+        impl.save_assemblies_from_fastas(context, params)
+    assert_exception_correct(got.value, ValueError(f"Cannot evaluate the string {var_name}"))
+
+    config['file_config']["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "-1"
+    impl = AssemblyUtil(config['file_config'])
+    with raises(Exception) as got:
+        impl.save_assemblies_from_fastas(context, params)
+    assert_exception_correct(got.value, ValueError(f"{var_name} cannot be negative"))
+
+
+def test_generator_overflow(config, scratch):
+    tmp_dir = scratch / ("test_generator_overflow" + str(uuid.uuid4()))
+    os.makedirs(tmp_dir)
+    data = Path('data')
+    file_names = [
+        'GCA_002506415.1_ASM250641v1_genomic.fna.gz',
+        'GCF_000007065.1_ASM706v1_genomic.fna.gz',
+        'GCF_000970165.1_ASM97016v1_genomic.fna.gz',
+        'GCF_000970185.1_ASM97018v1_genomic.fna.gz',
+        'GCF_000970205.1_ASM97020v1_genomic.fna.gz',
+        'GCF_000970245.1_ASM97024v1_genomic.fna.gz'
+    ]
+
+    # copy 6 assembly files into the data dir
+    for file_name in file_names:
+        shutil.copy(data / file_name, tmp_dir)
+
+    params = {
+        'workspace_id': config['ws_id'],
+        'inputs': [
+            {'file': tmp_dir / 'GCA_002506415.1_ASM250641v1_genomic.fna.gz', 'assembly_name': 'GCA_002506415.1_ASM250641v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000007065.1_ASM706v1_genomic.fna.gz', 'assembly_name': 'GCF_000007065.1_ASM706v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970165.1_ASM97016v1_genomic.fna.gz', 'assembly_name': 'GCF_000970165.1_ASM97016v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970185.1_ASM97018v1_genomic.fna.gz', 'assembly_name': 'GCF_000970185.1_ASM97018v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970205.1_ASM97020v1_genomic.fna.gz', 'assembly_name': 'GCF_000970205.1_ASM97020v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970245.1_ASM97024v1_genomic.fna.gz', 'assembly_name': 'GCF_000970245.1_ASM97024v1_genomic.fna.gz'}
+        ]
+    }
+
+    dfu = DataFileUtil(config['callback_url'], token=config['token'])
+    fta = FastaToAssembly(dfu, scratch)
+    fta.import_fasta_mass(params, 1, 10, 38300)
+
+
+def test_invalid_max_cumsize(config, scratch):
```

Comment:
This test could go in unit tests and be quite a bit simpler since checking the max cumsize happens very early in the call stack

Reply:
If this test can could go in unit tests why not overflow? They are all using FastaToAssembly directly. What criteria you use when deciding which goes which?

Reply:
I guess since you can now turn parallelization off you could do that, mock DFU, and check that you're seeing the expected inputs to DFU in unit tests. Either I didn't think of it before or maybe I said that when parallelization was always on and so mocking DFU would cause the picklization fail

Reply:
Not sure if it's simpler. It's easier to pass in max cumsize in integration test than unit test. It could be done, but still need to call _run_test_spec_fail() 3 times with each test_spec list length of 1. What we currently have seems more straightforward to me. Please let me know if you think differently.

Reply:
Unit tests are simpler because they only involve the code under test, and therefore are easier to understand and debug since there's less code involved. My rule is that if a test can be written as a unit tests it should be.

Reply:
Moved to unit tests. It is actually simpler than I expected.

Reply:
Yep, unit tests should almost always be simpler than integration tests. if they aren't, it might be a sign that the code under test needs refactoring

Reply:
: